### PR TITLE
legit: write commit message in a buffer

### PR DESCRIPTION
**Pull request by vindarel**
Currently, we can only write one-line commit messages in Lem/legit :S This improves the situation, we create a special buffer to write commit messages.

I only tested this evening while working on it.

### CODE REVIEW

**Comment by cxxxr** [comment]
I think this is a good change.
Is it still a draft PR?

**Comment by vindarel** [comment]
It is ready for those who want to keep testing lem/legit, but by looking at it I see a couple inline tests that you might not like :D (I like them… these will also be useful again shortly)

**Comment by cxxxr** [comment]
Yeah, I think the inline comment are not bad.
The only reason I thought it was a draft is that the new file was not added to asd and not loaded.

**Comment by vindarel** [comment]
hello, ping?

File: src/ext/legit/legit-commit.lisp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+" (clean-commit-message
+"one
+
+two
+
+# Please enter the commit message for your changes. Lines starting
+# with '#' will be ignored, and an empty message aborts the commit.
+")))
+
+(defun commit-continue ()
```
Comment (cxxxr) [inline src/ext/legit/legit-commit.lisp:90]: This is bound to C-c C-c, but it is not defined using define-command, so it cannot be called.

File: src/ext/legit/legit-commit.lisp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+         (lem-core/commands/window:previous-window)
+         ;; and refresh.
+         (legit-status))))))
+
+(define-command commit-abort () ()
+  (kill-buffer "*legit-commit*")
+  (lem-core/commands/window:previous-window))
+
+(define-command commit-abort-yes-or-no () ()
+  ;; TODO: prompt for confirmation.
```
Comment (cxxxr) [inline src/ext/legit/legit-commit.lisp:112]: Can we use prompt-for-y-or-n-p?

**Review by cxxxr — COMMENTED** [review]
This feature is very nice!
I tried it for a while and commented on a few things that caught my attention.

File: src/ext/legit/legit-commit.lisp (hunk 2 above)
Comment (vindarel) [reply src/ext/legit/legit-commit.lisp:112]: \+1 good catch thanks. Do we have a prompt where Y is the default choice? (so we can only press Enter, not "y")

**Review by cxxxr — APPROVED** [review]
LGTM

**Pull request merged**